src/Eshopworld.Messaging/TopicAdapter.cs

```diff
@@ -124,5 +129,19 @@ protected override void Dispose(bool disposing)
 
             base.Dispose(disposing);
         }
+
+        /// <inheritdoc />
+        protected override void RebuildReceiver()

```

Comment:
Can you change the signature of this to return the task and not block (await here). If you need this for the constructor put the Wait there. Then for any case where it needs to be rebuilt during application execution (after construction) it would be done async and not sync.